djpyfs/djpyfs.py

```diff
@@ -1,53 +1,57 @@
-'''
-
-This is a thin veneer around a pyfilesystem. It adds a few bits of
+"""
+This is a thin veneer around a `pyfilesystem`. It adds a few bits of
 functionality:
 
 1) Django configuration. This can go to Amazon S3 or a static
 filesystem.
 2) The ability to get URLs for objects stored on the filesystem.
 3) The ability to create objects with a limited lifetime. A 
 task can garbage-collect those objects. 
+"""
 
-'''
 from __future__ import absolute_import
 
 import os
 import os.path
 import types
 
+from boto.s3.connection import S3Connection
 from django.conf import settings
 from fs.osfs import OSFS
+from fs.s3fs import S3FS
 
 from .models import FSExpirations
 
 
 if hasattr(settings, 'DJFS'):
-    djfs_settings = settings.DJFS  #pragma: no cover
+    DJFS_SETTINGS = settings.DJFS  # pragma: no cover
 else:
-    djfs_settings = {'type' : 'osfs',
-                     'directory_root' : 'django-pyfs/static/django-pyfs', 
-                     'url_root' : '/static/django-pyfs'}
+    DJFS_SETTINGS = {'type': 'osfs',
+                     'directory_root': 'django-pyfs/static/django-pyfs', 
+                     'url_root': '/static/django-pyfs'}
 
-s3conn = None
+# Global to hold the active S3 connection. Prevents needing to reconnect
+# several times in a request. Connections are set up below in `get_s3_url`.
+S3CONN = None
 
 
 def get_filesystem(namespace):
-    ''' Returns a pyfilesystem for static module storage. 
-
-    The file system will have two additional properties: 
-    1) get_url: A way to get a URL for a static file download
-    2) expire: A way to expire files (so they are automatically destroyed)
-    '''
-    if djfs_settings['type'] == 'osfs':
-        return get_osfs( namespace )
-    elif djfs_settings['type'] == 's3fs':
-        return get_s3fs( namespace )
+    """
+    Returns a patched pyfilesystem for static module storage based on
+    `DJFS_SETTINGS`.
+    """
```

Comment:
I think, to someone unfamiliar with the code, it's helpful to document what's patched about the file system, as in the original comment.